### DISCUSSION TITLE
extism with IoT devices

### DISCUSSION THREAD

**Discussion by TAOFOR4** [extism/extism#713] [Q&A]
Hi, thx for this interesting project. I saw there mentions IoT in README file. I am wondering which area of IoT extism works with. Can it be used with IoT devices like raspberry pico or ESP32 and so on? Thank you!

**Comment by zshipko** [comment]
We just published our [wamr-sdk](https://github.com/extism/wamr-sdk) which supports arm and esp32, it's still under development so any feedback is appreciated!

**Reply by TAOFOR4** [reply]
Hi thx for reply, I have related devices and will try the project soon. Also, I am new in this area so this question might be a little stupid. Considering extism using wamr as backend, then what's its advantage compared with using wamr directly?

**Reply by TAOFOR4** [reply]
Also, do you guys have/recommend some suitable testing examples or benchmarks that I can try? I have ESP32 C6 and pico which are supported, I guess. (I haven't seen any on [wamr-sdk](https://github.com/extism/wamr-sdk)

**Reply by zshipko** [reply]
Extism implements a slightly higher-level interface over WAMR, which does provide some nice ways of getting data in and out of memory compared to wasmtime, but we also have several PDKs to make writing Extism compatible plugins easier.

I just published the wamr-sdk this week so I still don't have many examples. I was planning on putting together something for esp32 next week and can post an update here when I get something working!
